### PR TITLE
Migrate Subscriptions to Extensibility

### LINKED ISSUE #1866: EXTENSIBILITY: Migrate Subscriptions to extensibility
Description

Move Subscriptions functionality to an extension.

**Acceptance Criteria**

- [ ] The application behaves and looks as similar as possible - changes are accepted
- [ ] Test are adjusted
- [ ] We try to reuse the existing widgets. If new widgets are needed, they need to be created and documented
- [ ] https://github.com/kyma-project/busola/issues/1832
- [ ] https://github.com/kyma-project/kyma/issues/15586

### PR DESCRIPTION
**Pull request by mrCherry97**
**Description**

Changes proposed in this pull request:

- add subscription config map,
- small fix in ResourceLink,
- add a template for GenericList widget in form,
- add `readableTimestamp` function,
- improve docs.

**Related issue(s)**
Closes #1866

### CODE REVIEW

File: docs/extensibility/form-section.md [hunk 1]
```diff
@@ -398,6 +399,21 @@ GenericList widgets render an array as a list of collapsible sections with their
...
+       placeholder: placeholder.eventType
+     - path: '[].eventSource.value'
+   template:
+     eventSource:
+     property: source
+       type: exact
+       value: ''
+     eventType:
+       property: type
+       type: exact
```
Comment (Lyczeq) [inline docs/extensibility/form-section.md:416]: Could you add one tab before `property`, please? I believe it should be at the same level as `type` and `value`.

File: examples/resources/configuration/subscriptions.yaml [hunk 2]
```diff
@@ -0,0 +1,137 @@
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: subscriptions
+  namespace: kube-public
+  labels:
+    app.kubernetes.io/name: subscriptions.eventing.kyma-project.io
```
Comment (Lyczeq) [inline examples/resources/configuration/subscriptions.yaml:7]: Hm, shouldn't `name` and `app.kubernetes.io/name` label be the same? ConfigMap automatically adds that label based on the name. We have a similar approach with other extensibility configurations.

File: examples/resources/configuration/subscriptions.yaml [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+                    name: spec.filter.event.value
+      - widget: EventList
+        filter: '$matchEvents($item, $root.kind, $root.metadata.name)'
+        name: events
+        defaultType: NORMAL
+        hideInvolvedObjects: true
+  form: |-
+    - path: spec.sink
+      name: spec.sink
+      required: true
```
Comment (Lyczeq) [inline examples/resources/configuration/subscriptions.yaml:89]: I don't think that `required` flag is needed here. It's automatically set based on the schema.

File: examples/resources/configuration/subscriptions.yaml [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+      simple: true
+    - widget: GenericList
+      path: spec.filter.filters
+      simple: true
+      children:
+        - path: '[].eventType.value'
+          name: spec.filter.eventType
+          placeholder: placeholder.eventType
+          required: true
```
Comment (Lyczeq) [inline examples/resources/configuration/subscriptions.yaml:98]: I don't think that `required` flag is needed here. It's automatically set based on the schema.

**Review by Lyczeq — CHANGES_REQUESTED** [review]

**Review by grego952 — APPROVED** [review]

File: examples/resources/configuration/subscriptions.yaml [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+      source: 'status.ready = false ? "error" : "ready"'
+      widget: Badge
+      description: status.conditions.message
+    - name: status.cleanEventTypes
+      source: status.cleanEventTypes
+      widget: Labels
+  translations: |-
+    en:
+      inputInfo.sink: Sink structure, 'http://{SERVICE}.{NAMESPACE}.svc.cluster.local'
+      inputInfo.eventType: Event Type structure, 'sap.kyma.custom.{APP}.{EVENT.NAME}.{VERSION}'
```
Comment (grego952) [inline examples/resources/configuration/subscriptions.yaml:123]:
```suggestion
      inputInfo.eventType: Event Type structure, 'sap.kyma.custom.{APP}.{NAME}.{VERSION}'
```

**Review by grego952 — APPROVED** [review]

**Review by Lyczeq — APPROVED** [review]

**Pull request merged**